source/common/common/BUILD

```diff
@@ -262,11 +279,21 @@ envoy_cc_library(
 envoy_cc_library(
     name = "version_lib",
     srcs = ["version.cc"],
+    hdrs = select({
+        "//bazel:manual_stamp": [":generate_version_linkstamp"],
+        # By default the header file is empty.
+        # This is done so that the definitions linked via the linkstamp rule don't cause collisions.
+        "//conditions:default": [":generate_version_linkstamp_empty"],
+    }),
     copts = envoy_select_boringssl(
         ["-DENVOY_SSL_VERSION=\\\"BoringSSL-FIPS\\\""],
         ["-DENVOY_SSL_VERSION=\\\"BoringSSL\\\""],
     ),
     linkstamp = "version_linkstamp.cc",
```

Comment:
@mattklein123 as I have implemented here, this PR does not solve for #2181 for two reasons:
1. The generated header still takes the value from the workspace status.
2. The bazel rule does not conditionally add the `linkstamp` attribute, so using `--define manual_stamp=manual_stamp` would actually cause duplicated definitions.

I am starting to think that the best way to resolve #2181 is to include a `SOURCE_VERSION` file and make that documented per the issue, or include it in the tarball somehow.

However, I am open to fixing by more bazel work here.

Reply:
cc @dio